File: higher-of-two-rolls.py
import random
import time

number_of_sides  =  int(input("How many sides on your dice? "))

t1 = time.time()

sum_of_results = 0.0

trials = 0
while trials < 10**6:
    sum_of_results += max([int(random.random()*number_of_sides)+1,int(random.random()*number_of_sides)+1])
    trials += 1

print("Average result of rolling two and taking the highest is about {0}".format(sum_of_results/trials))
print(time.time() - t1)

File: higher-of-two-rolls.cpp
#include <iostream>
#include <random>
#include <ctime>

using namespace std;

int main() {
    int number_of_sides;
    cout << "How many sides on your dice? ";
    cin >> number_of_sides;

    double t1 = clock();

    float sum_results = 0;

    int trials = 1000000;

    for (int i = 0; i < trials; i++) {
        sum_results += max(rand() % number_of_sides + 1, rand() % number_of_sides + 1);
    }

    cout << "Average result of rolling two and taking the highest is " << sum_results / trials << endl;
    cout << "Time taken: " << (clock() - t1) / CLOCKS_PER_SEC << " seconds" << endl;
}